runtime/src/main/cpp/Memory.cpp

```diff
@@ -2504,6 +2567,99 @@ void shareAny(ObjHeader* obj) {
   container->makeShared();
 }
 
+OBJ_GETTER0(detectCyclicReferences) {
+  // Collect rootset, hold references to simplify remaining code.
+  KRefList rootset;
+  lock(&g_leakCheckerGlobalLock);
+  auto* candidate = g_leakCheckerGlobalList;
+  while (candidate != nullptr) {
+    CreateStablePointer(candidate);
```

Comment:
Looks much like an abuse of `CreateStablePointer`. Why not use `AddRef` directly instead?

Reply:
Well, can do it either way.